### DISCUSSION TITLE
Nginx Proxy & getRemoteAddress

### DISCUSSION THREAD

**Discussion by Zanoro9000** [uNetworking/uWebSockets.js#702] [Q&A] [answered]
Hello,

Enjoying the library so far, its very fast and efficient while being easy to use. 

I have a problem where I have a Nginx proxy running in front of my server, and I am losing the ip address of the remote request on the websocket itself. `.getProxiedRemoteAddress` works on my http requests, but I am always getting `127.0.0.1` for `.getRemoteAddress` on the websocket instance. Is there a way to "trick" the request into having the correct address using nginx, or another way to fetch that address on the websocket?

My initial thought is to store it on the ws during the upgrade, but if there's an easier way that would be great!

Thanks!

**Comment by e3dio** [comment, marked as answer]
You need to use `upgrade` websocket handler, you can use `res.getProxiedRemoteAddress` and store result on ws object, see https://github.com/uNetworking/uWebSockets.js/discussions/339#discussioncomment-172632

**Reply by Zanoro9000** [reply]
Okay, that makes sense! I was using the default upgrade until now but if that has to be done then that's fine. Thanks!
